src/engine/strat_engine/range_alloc.rs

```diff
@@ -57,44 +97,58 @@ impl RangeAllocator {
         for &(off, len) in to_free {
             self.check_for_overflow(off, len);
 
-            for val in *off..*off + *len {
-                let removed = self.used.remove(&val);
-                if removed == false {
-                    panic!(format!("tried to remove value {} not present in RangedSet", val));
+            let (prev_off, prev_len) = self.used
+                .range((Unbounded, Included(off)))
+                .rev()
+                .next()
+                .map(|(k, v)| (*k, *v))
+                .expect("must match an existing range");
```

Comment:
I don't see how that differs in function from the PR's current implementation: If we don't find an existing range, panic. Can you elaborate?